src/server/project.js

```diff
@@ -0,0 +1,348 @@
+    if (geolocationIndex >= 0) {
+        rawProjectPluginList.splice(geolocationIndex, 1);
+    }
+
+    var simulatedProjectPlugins = {};
+
+    rawProjectPluginList.forEach(function (pluginId) {
+        var pluginFilePath = pluginUtil.findPluginPath(this.projectRoot, pluginId);
+
+        if (pluginFilePath && pluginUtil.shouldUsePluginWithDebugHost(pluginFilePath, true, debugHostHandlers)) {
+            simulatedProjectPlugins[pluginId] = pluginFilePath;
+
+            if (pluginFilePath.indexOf(dirs.plugins) === 0) {
+                this.pluginsTelemetry.simulatedBuiltIn.push(pluginId);
+            } else {
+                this.pluginsTelemetry.simulatedNonBuiltIn.push(pluginId);
+            }
+        } else {
+            this.pluginsTelemetry.notSimulated.push(pluginId);
+        }
+    }.bind(this));
+
+    // Find built-in debug-host plugins (plugins that should be part of the simulation due to the presence of a
+    // debug-host, even if they weren't added to the project).
+    var simulatedDebugHostPlugins = {};
+
+    if (debugHostHandlers) {
+        var rawBuiltInPluginList = utils.getDirectoriesInPath(dirs.plugins);
+
+        rawBuiltInPluginList.forEach(function (pluginId) {
+            if (simulatedDefaultPlugins[pluginId] || simulatedProjectPlugins[pluginId]) {
+                return;
+            }
+
+            var pluginPath = path.join(dirs.plugins, pluginId);
+
+            if (pluginPath && pluginUtil.shouldUsePluginWithDebugHost(pluginPath, null, debugHostHandlers)) {
+                simulatedDebugHostPlugins[pluginId] = pluginPath;
+                this.pluginsTelemetry.simulatedBuiltIn.push(pluginId);
+            }
+        }.bind(this));
+    }
+
+    // Register the plugins. The default plugins are first, then the debug-host plugins (to have the correct order in
+    // sim-host).
+    var that = this;
+    function registerPlugins(pluginDictionary) {
+        Object.keys(pluginDictionary).forEach(function (pluginId) {
+            that.plugins[pluginId] = pluginDictionary[pluginId];
+        });
+    }
+
+    registerPlugins(simulatedDefaultPlugins);
+    registerPlugins(simulatedDebugHostPlugins);
+    registerPlugins(simulatedProjectPlugins);
+
+    this._simulator.telemetry.sendTelemetry('plugin-list', {
+        simulatedBuiltIn: this.pluginsTelemetry.simulatedBuiltIn
+    }, {
+        simulatedNonBuiltIn: this.pluginsTelemetry.simulatedNonBuiltIn,
+        notSimulated: this.pluginsTelemetry.notSimulated
+    });
+    this._addPlatformDefaultHandlers();
+    this._populateRouter();
+};
+
+Project.prototype.getPlugins = function () {
+    return this.plugins;
+};
+
+Project.prototype.getRouter = function () {
+    this._router = this._router || cordovaServe.Router();
+    return this._router;
+};
+
+Project.prototype.hasBuiltInPluginTelemetry = function (plugin) {
+    var builtIn = this.pluginsTelemetry.simulatedBuiltIn;
+
+    return (builtIn.indexOf(plugin) !== -1);
```

Comment:
@guillaumejenkins I've added this new function that checks if the given plugin has telemetry registered (built-in plugin). This logic was implemented in the telemetry module, when checking if the telemetry should be anonymized